content/blog/2020-12-19-LLVM-IR-to-assembly.md

```diff
@@ -0,0 +1,77 @@
+| SSE                                     | 144.06                | 124.48 | 124.33       |
+| AV_4mem                                 | 140.11                | 170.03 | 170.14       |
+| AVX_8                                   | 93.48                 | 98.99  | 95.75        |
+
+### Results and Analysis
+As this microbenchmark is designed to measure the maximum performance of the complied code, I have chosen to aggregate the result a min function. Preliminary analysis has shown that reducing the number of architectural registers has little or no effect on the performance on microbenchmarks.  This is as expected as microbenchmarks do not have high amounts register pressure as they often contain couple of register operands.  In this case, the complier backend implementation has more of an impact on the final performance then the modification on the number of x86-64 architectural registers.  
+
+
+## FFT benchmark:
+
+### Process and Methology
+The benchmark preforms a set of FET calculations.  A total of 6 interactions have been performed and the data was preformed statistical analysis.  Execution time is measured in wall-clock time.
+
+### Results
+Table 2, FET benchmark in wall-clock time,average number cycles required to complete the operation.  
+| Exec time (s) | gcc         | clang       | clang (0.75) |
+|---------------|-------------|-------------|--------------|
+| Average       | 2833314     | 2548542     | 3081897      |
+| Min           | 2522625     | 2456454     | 2643296      |
+| Std. Dev      | 381923      | 66116       | 497256       |
+| CI-Lower(0.01)| 2431691     | 2479015     | 2558993      |
+| CI-Upper(0.01)| 3234936     | 2618069     | 3604801      |
+| CI-Lower(0.05)| 2527717     | 2495639     | 2684016      |
+| CI-Upper(0.05)| 3138911     | 2601446     | 3479778      |
+
+### Results and Analysis
+In the FFT benchmark, the modified version with fewer registers consistently underperforms unmodified versions of GCC and Clang.  However, something of note worth is the minimum time needed for execution is relatively close for GCC and modified Clang version. At 0.01 significant the confidence interval overlaps between the categories clang and modified Clang , therefore we conclude the performance differences between group is not statistically significant. However, at 0.05 significance the results is statistically significant between original Clang and the modified Clang implementation.  In both cases, the performance difference between GCC and Clang are not statistically significant.  Despite the needed to have a greater sample size to have statically significant numbers for 0.01 significance level, and the need of a more comprehensive benchmark to identify the patterns and implications of reducing the number of architectural registers.  I have nevertheless demonstrated the ability to modify LLVM to generate assembly with a sub-set of all the original ISA registers.  
```

Comment:
the performance differences between group is not statistically significant: is -> are; add "the" before "group"